flow/peerdbenv/env.go

```diff
@@ -86,7 +87,15 @@ func decryptWithKMS(ctx context.Context, data []byte) ([]byte, error) {
 	return decrypted.Plaintext, nil
 }
 
+var KMSCache sync.Map
+
 func GetEnvBase64EncodedBytes(ctx context.Context, name string, defaultValue []byte) ([]byte, error) {
+	if cacheVal, ok := KMSCache.Load(name); ok {
```

Comment:
There is a slight race here where two concurrent accesses will run in parallel. This is fine